src/vos/vos_obj.c

```diff
@@ -1826,12 +1826,14 @@ vos_obj_iter_aggregate(daos_handle_t ih, bool discard)
 			 * at present prevent underpunch
 			 */
 			D_ERROR("Removing orphaned single value tree\n");
+			D_ASSERT(0);
```

Comment:
Maybe change these conditionals to a D_ASSERTF() ... the comment above leads to believe that was the initial intent.

Reply:
Ok, cleaned it up.